aip/auth/4115.md

```diff
@@ -5,124 +5,115 @@ state: approved
+### Access Default mTLS Credentials
 
-If the application runs on a Compute Engine instance (or equivalent runtime),
-the auth library can request the following URL to get an access token associated
-with instance:
+To leverage Google’s default mTLS credentials, the client **should** retrieve configurations from 
+MDS. The MDS in all virtual environments (GCE, Serverless, and GKE) expose an HTTP endpoint that 
+serves a configuration that specifies how to access Google's default mTLS credentials. This endpoint 
+is called the mTLS configuration endpoint.
 
+The URL of the MDS's mTLS configuration endpoint is: 
 ```
-http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token
+http://metadata.google.internal/computeMetadata/v1/googleAutoMtlsConfiguration
 ```
 
-In response, the auth library will usually receive a token in the following JSON
-format:
+The request to the MDS's mTLS configuration endpoint **should** be an HTTP GET request without any 
+parameter or payload.
 
-```json
-{
-      "access_token":"YOUR_ACCESS_TOKEN",
-      "expires_in":3599,
-      "token_type":"Bearer"
- }
-```
+The response from the MDS's mTLS configuration endpoint **should** contain the following 
+information:
 
-#### Scopes
-Access tokens obtained from the metadata server contain the scopes specified at instance creation.
-For example, if an instance is granted only the
-https://www.googleapis.com/auth/storage-full scope for Cloud Storage, then the tokens obtained 
-from the instance's metadata server have only the `storage-full` scope.
+* The **Secure Session Agent** address: the client doesn’t have direct access to mTLS credentials. 
+  The Secure Session Agent manages default mTLS credentials. The client can only use mTLS 
+  credentials through the Secure Session Agent. The address can be an IP:port address or a file path 
+  as a Unix Domain Socket (UDS).
 
-When running on GCE, the metadata server ignores requests for custom scopes.
-On GCE equivalent runtimes, clients can request a different set of scopes to the metadata server
-using the `scopes` url parameter, e.g.:
+The client **must** follow the steps below to access Google’s default mTLS credentials.
 
-```
-http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token?scopes=comma-separated-list-of-scopes
-```
-
-The auth library **should** allow the caller to optionally specify a list of custom scopes,
-and add the `scopes` parameter to the request when needed.
-Depending on the runtime environment, the request for custom scopes will be transparently
-ignored by the server (GCE) or fulfilled (GCE equivalent).
+1. Check if the remote endpoint supports mTLS. 
+   * If yes, go to step (2).
+   * If not, go to step (3). 
+2. Send a request to the MDS's mTLS configuration endpoint. If the request is successful and the 
+   response contains a Secure Session Agent address, use the address to access Google default mTLS
```

Comment:
micro-nit: s/Google/Google's

For consistency throughout the AIP.

Reply:
Done.